tests/unit/test_artifactory_path.py

```diff
@@ -751,6 +737,129 @@ def test_auth(self):
         a = P("http://a/artifactory/", auth=("foo", "bar"))
         self.assertEqual(a.auth, ("foo", "bar"))
 
+    @responses.activate
+    def test_deploy_file(self):
+        """
+        Test that file uploads to the path
+        :return:
+        """
+        P = self.cls
+        path = P(
+            "http://artifactory.local/artifactory/libs-release-local",
+            auth=("foo", "bar"),
+        )
+
+        test_file = pathlib.Path("test.zip")
+        # file is required to calculate checksums
+        with open(test_file, "w") as file:
+            file.write("I am a test file")
+
+        constructed_url = (
+            "http://artifactory.local/artifactory" "/api/storage" "/libs-release-local"
+        )
+        responses.add(
+            responses.GET,
+            constructed_url,
+            status=200,
+            json=self.dir_stat,
+        )
+
+        matrix_parameters = "deb.architecture=amd64;deb.component=contrib;deb.distribution=dist1;deb.distribution=dist2"
+        constructed_url = f"{path}{test_file};{matrix_parameters}"
+        responses.add(
+            responses.PUT, constructed_url, status=200, match_querystring=True
+        )
+
+        try:
+            path.deploy_file(
+                str(test_file),
+                explode_archive=True,
+                explode_archive_atomic=True,
+                parameters={
+                    "deb.architecture": "amd64",
+                    "deb.component": "contrib",
+                    "deb.distribution": ["dist1", "dist2"],
+                },
+            )
+        finally:
+            test_file.unlink()
+
+        request_url = responses.calls[1].request.url
+        self.assertEqual(request_url, constructed_url)
+
+        # verify that all headers are present and checksums are calculated properly
+        headers = responses.calls[1].request.headers
+        self.assertIn("X-Checksum-Md5", headers)
```

Comment:
Can we combine asserts into one?

Like this
```python

expected_header = {
    "X-Checksum-Sha1, "d999740bb5423b9defcb9525245f471c37eb4382",
    "X-Checksum-Sha256",
            "421a9d115635ef05cf423b92de6a2285139c9b77c231d2f5027a0e42b99dc800"
}
checked_headers = {x:y for x,y in headers.items() if x in expected_headers}
self.assertEqual(checked_header, expected_headers)
```

Reply:
`assertDictEqual` instead of `assertEqual`

Or `assert` if we use pytest :)

Reply:
@allburov 
I thought about this, but I would prefer to have them separate. Then in case if somebody corrupts of the the header, then you can easily understand what happened

but if you do a bulk comparison, then you have to run a debugger to understand when it was raised

Reply:
eg 
when I changed handling of the file via Temp all checksums where changed

then in PyCharm it has shown me the line that failed immediately with link to the line number. Then I can easily understand that MD5 was present in header but was changed 

I vote for usability against short code

Reply:
With pycharm and pytest dictionary comparisons work like a charm, let's use it

![image](https://user-images.githubusercontent.com/4376814/132309016-ad6f05b6-397a-4a3f-a1bf-2b8c4be900fe.png)

Reply:
We can come back to it later, never mind